packages/dev/core/src/XR/features/WebXRDepthSensing.ts

```diff
@@ -0,0 +1,320 @@
+     */
+    public get latestDepthImageTexture(): RawTexture | null {
+        if (!this._cachedDepthImageTexture) {
+            return null;
+        }
+        return this._cachedDepthImageTexture;
+    }
+
+    /**
+     * XRWebGLBinding which is used for acquiring WebGLDepthInformation
+     */
+    private _glBinding?: XRWebGLBinding;
+
+    /**
+     * The module's name
+     */
+    public static readonly Name = WebXRFeatureName.DEPTH_SENSING;
+
+    /**
+     * The (Babylon) version of this module.
+     * This is an integer representing the implementation version.
+     * This number does not correspond to the WebXR specs version
+     */
+    public static readonly Version = 1;
+
+    /**
+     * Creates a new instance of the depth sensing feature
+     * @param _xrSessionManager the WebXRSessionManager
+     * @param options options for WebXR Depth Sensing Feature
+     */
+    constructor(_xrSessionManager: WebXRSessionManager, public readonly options: IWebXRDepthSensingOptions) {
+        super(_xrSessionManager);
+        this.xrNativeFeatureName = "depth-sensing";
+
+        // https://immersive-web.github.io/depth-sensing/
+        Tools.Warn("depth-sensing is an experimental and unstable feature.");
+    }
+
+    /**
+     * attach this feature
+     * Will usually be called by the features manager
+     *
+     * @returns true if successful.
+     */
+    public attach(force?: boolean | undefined): boolean {
+        if (!super.attach(force)) {
+            return false;
+        }
+
+        const isBothDepthUsageAndFormatNull = this._xrSessionManager.session.depthDataFormat == null || this._xrSessionManager.session.depthUsage == null;
+        if (isBothDepthUsageAndFormatNull) {
+            return false;
+        }
+
+        this._glBinding = new XRWebGLBinding(this._xrSessionManager.session, this._xrSessionManager.scene.getEngine()._gl);
+
+        return true;
+    }
+
+    /**
+     * Dispose this feature and all of the resources attached
+     */
+    public dispose(): void {
+        this._cachedDepthImageTexture?.dispose();
+    }
+
+    protected _onXRFrame(_xrFrame: XRFrame): void {
+        const referenceSPace = this._xrSessionManager.referenceSpace;
+
+        const pose = _xrFrame.getViewerPose(referenceSPace);
+        if (pose == null) {
+            return;
+        }
+
+        for (const view of pose.views) {
+            switch (this.depthUsage) {
+                case "cpu-optimized":
+                    this._updateDepthInformationAndTextureCPUDepthUsage(_xrFrame, view, this.depthDataFormat);
+                    break;
+
+                case "gpu-optimized":
+                    if (!this._glBinding) {
+                        break;
+                    }
+
+                    this._updateDepthInformationAndTextureWebGLDepthUsage(this._glBinding, view, this.depthDataFormat);
+                    break;
+
+                default:
+                    Tools.Error("Unknown depth usage");

```

Comment:
shouldn't we detach if that happens? otherwise this will be logged on each frame